phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -1741,6 +1743,13 @@ public MutationState createIndex(CreateIndexStatement statement, byte[][] splits
         if (connection.getSCN() != null) {
             return buildIndexAtTimeStamp(table, statement.getTable());
         }
+
+        String dataTableFullName = SchemaUtil.getTableName(
+                tableRef.getTable().getSchemaName().getString(),
+                tableRef.getTable().getTableName().getString());
```

Comment:
@kadirozde - curious why the refresh is needed here? The tableRef only seems to be used to get the base table name to give to the ServerBuildIndexDDLCompiler

Reply:
I needed this for the initial version of the compiler where the index table was searched in the index list of the PTable for the data table. MetadataClient constructs the PTable object for the data table before the index table is created and thus the object does have a reference to the newly created index. The complier has changed and this search was eliminated. So there is no need to refresh the PTable object anymore. I will remove the refresh code.

Reply:
Thanks for explaining, @kadirozde . Once the refresh code gets removed, I think this is ready to go so I'll +1 and commit.

Reply:
@gjacoby126, I have removed the refresh code. Thank you for discovering this issue, helping me understand the code, and reviewing the patch!